### ISSUE TITLE
Fail to build UTs for PolyType when disabling FW_SERIALIZABLE_TO_STRING

### ISSUE THREAD

**Issue by kevin-f-ortega** [nasa/fprime#1727]
| | |
|:---|:---|
|**_F´ Version_**| e47cc1abc8e83f763861291919c9f6b63e8aeaa2|
|**_Affected Component_**|   |
---
## Problem Description

Fail to build UTs for PolyType when disabling FW_SERIALIZABLE_TO_STRING

## How to Reproduce

1. Disable FW_SERIALIZABLE_TO_STRING in config/FpConfig.hpp
2. Run fprime-util check --all -j 4

## Expected Behavior

I expect UTs to build correctly.

**Comment by LeStarch** [comment]
There are several issues.  These need to be fixed in hand-code and FPP code. 

1. `#include` of string.hpp does not use || BUILD_UT
2. BUILD_UT should likely be (defined(BUILD_UT) && BUILD_UT) as BUILD_UT is not always set
3. `FW_SERIALIZABLE_TO_STRING_BUFFER_SIZE` is used when it is only defined by FW_SERIALIZABLE_TO_STRING=1

All three issues need to be fixed.

Related to #1705, #1766

**Comment by LeStarch** [comment]
Blocked by: https://github.com/fprime-community/fpp/issues/369

**Comment by matt392code** [comment]
**String Header Include Issue:**
// Current problematic include (presumably)
#include <Fw/Types/String.hpp>
// Should be modified to:
#if defined(FW_SERIALIZABLE_TO_STRING) || defined(BUILD_UT)
#include <Fw/Types/String.hpp>
#endif

**BUILD_UT Macro Check Enhancement:**
// Current potentially problematic check
#ifdef BUILD_UT
// or
#if BUILD_UT
// Should be modified to more robust:
#if defined(BUILD_UT) && BUILD_UT

**Buffer Size Macro Usage:**
// Current problematic usage
#ifdef FW_SERIALIZABLE_TO_STRING
#define FW_SERIALIZABLE_TO_STRING_BUFFER_SIZE 256  // example size
#endif
// Usage needs to be wrapped:
#if defined(FW_SERIALIZABLE_TO_STRING) && FW_SERIALIZABLE_TO_STRING
... use FW_SERIALIZABLE_TO_STRING_BUFFER_SIZE ...
#endif

**Implementation Plan:**
 - Locate all instances of String.hpp includes in PolyType-related files
 - Update conditional compilation guards around string operations
 - Review and update all BUILD_UT conditionals to use the more robust check
 - Ensure buffer size usage is properly guarded
 - Add unit tests to verify behavior both with and without FW_SERIALIZABLE_TO_STRING enabled
 
This fix ensures that:
 - Unit tests can build regardless of FW_SERIALIZABLE_TO_STRING setting
 - Proper header inclusion based on build configuration
 - More robust macro checking
 - Correct conditional usage of buffer size configurations

**Comment by bocchino** [comment]
Superseded by https://github.com/nasa/fprime/issues/2467.
